src/openforms/formio/components/custom.py

```diff
@@ -422,10 +425,28 @@ def validate(self, attrs):
 
         city = attrs.get("city", "")
         street_name = attrs.get("streetName", "")
+        postcode = attrs.get("postcode", "")
+
+        if postcode_regex := glom(
+            self.component, "ofComponents.postcode.validate.pattern", default=""
+        ):
+            if not re.fullmatch(postcode_regex, postcode):
```

Comment:
Should we do something here to avoid possible ReDoS attacks with specific patterns?

Reply:
hrm, is there a way to detect those easily?

There are plans to offer some "pre-baked" validations like house number/house letter that users can select in a dropdown, with a final option "custom" that would let them write the regex themselves (as they do now), so that should minimize the risk of accidentally writing a bad regex.

Reply:
I checked, there is a library `re2` by Google which is supposed to mitigate this https://github.com/google/re2/wiki/WhyRE2

There is also discussion in Django to use `re2` instead of `re`: https://forum.djangoproject.com/t/using-google-re2-instead-of-re-in-django/31841